### ISSUE TITLE
nullArgument false positive

### ISSUE THREAD

**Issue by bkdotcom** [vimeo/psalm#3309]
```php
if (isset($this->cfg['foo'])) {
    $this->bar->baz('arg1', $this->cfg['foo']);
}
```
results in
> Argument 2 of bar::baz cannot be null, null value provided to parameter with type callable

I don't see how arg2 could possibly be null since the call is wrapped in a "if not null" condition

Psalm version: 3.11.2@d47....

see https://psalm.dev/r/21c1082577

**Comment by bkdotcom** [comment]
not quite able to reproduce on psalm.dev... get
> Argument 2 of foo::bar cannot be mixed, expecting array<array-key, mixed>|callable
which can be avoided by additional conditions
